### PR TITLE
[Improve][connector-file] unifiy option between file source/sink and update document

### LINKED ISSUE #5461: [Feature][FileConnector] Unify some option in file source/sink connector
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

We have some option that is different in the file family connector.

For example, in the FtpFile connector sink, we support compress_codec, but we don't set compress_codec in the source, this might be strange.

And further more, in localfile/ftpfile/hdfsfile, there also exist some option that is different, e.g. some support row_delimiter, some don't support.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by liunaijie**
### Purpose of this pull request

close #5461 and #4572 
1. unified option in file source/sink (source parameter delimiter update to field_delimiter)
2. update the doucment
 a) update parameter name
 b) refactor docs using unified format

### Does this PR introduce _any_ user-facing change?
yes, the file source paramter delimiter updated to field_delimiter to keep same with sink paramter

### How was this patch tested?

use e2e 

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  4. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  5. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by Hisoka-X** [comment]
So after this PR, seem like `delimiter` will not worked. This is unacceptable and will hinder users who need to upgrade. If we need to modify a configuration key, we should be compatible with the old configuration at the same time.

**Comment by Hisoka-X** [comment]
So I think you should answer this question.
image

Also please update the title.

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/BaseSourceConfig.java [hunk 1]
```diff
@@ -40,8 +40,8 @@ public class BaseSourceConfig {
                     .noDefaultValue()
                     .withDescription("The file path of source files");
 
-    public static final Option<String> DELIMITER =
-            Options.key("delimiter")
+    public static final Option<String> FIELD_DELIMITER =
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/BaseSourceConfig.java:43]: Directly modifying parameter names can cause compatibility issues.
You can update option like this:
```
    public static final Option<String> FIELD_DELIMITER =
            Options.key("field_delimiter")
                    .stringType()
                    .defaultValue(TextFormatConstant.SEPARATOR[0])
                    .withFallbackKeys("delimiter")
                    .withDescription(
                            "The separator between columns in a row of data. Only needed by `text` file format");


```

And then you need update the code at the point used this option.

```
String delimiter =
                        ReadonlyConfig.fromConfig(pluginConfig)
                                .get(BaseSourceConfig.FIELD_DELIMITER);
```

**Review by EricJoy2048 — CHANGES_REQUESTED** [review]

**Review by Hisoka-X — COMMENTED** [review]
Could you add a test case for old config key `delimiter`? We should make sure the `delimiter` work fine too after this change.

File: docs/en/connector-v2/source/FtpFile.md [hunk 2]
```diff
@@ -221,9 +154,73 @@ connector will generate data as the following:
 |---------------|-----|--------|
 | tyrantlucifer | 26  | male   |
 
-### common options
+### delimiter/field_delimiter [string]
 
-Source plugin common parameters, please refer to [Source Common Options](common-options.md) for details.
+**delimiter** parameter will deprecate after version 2.3.5, please use **field_delimiter** instead.
```
Comment (Hisoka-X) [inline docs/en/connector-v2/source/FtpFile.md:159]:
```suggestion
**delimiter** parameter will deprecate after version 2.3.4, please use **field_delimiter** instead.
```

File: docs/en/connector-v2/source/HdfsFile.md [hunk 3]
```diff
@@ -55,9 +55,13 @@ Read data from hdfs file system.
...
 | compress_codec            | string  | no       | none                | The compress codec of files                                                                                                                                                                                                                                                                                                                   |
+| common-options            |         | no       | -                   | Source plugin common parameters, please refer to [Source Common Options](common-options.md) for details.                                                                                                                                                                                                                                      |
+
+### delimiter/field_delimiter [string]
+
+**delimiter** parameter will deprecate after version 2.3.5, please use **field_delimiter** instead.
```
Comment (Hisoka-X) [inline docs/en/connector-v2/source/HdfsFile.md:64]:
```suggestion
**delimiter** parameter will deprecate after version 2.3.4, please use **field_delimiter** instead.
```

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-file-local-e2e/src/test/resources/excel/local_excel_projection_to_assert_with_delimiter.conf [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+          field_type = boolean
+          field_value = [
+            {
+              rule_type = NOT_NULL
+            }
+          ]
+        }
+      ]
+    }
+  }
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-file-local-e2e/src/test/resources/excel/local_excel_projection_to_assert_with_delimiter.conf]: The test case seem like can not prove the delimiter is ":". Could you add a rule to check row value? You can refer https://github.com/apache/seatunnel/pull/5083/files#diff-a68b2797462655fd5ee139c5c129df223f4de9c25e72e41a6432012d00ede0a8

**Review by Hisoka-X — APPROVED** [review]

**Comment by Hisoka-X** [comment]
Thanks @liunaijie .Waiting CI pass.

**Review by EricJoy2048 — APPROVED** [review]

**Pull request merged**